### PR TITLE
add more tests

### PR DESCRIPTION
**Pull request by jw3126**

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -1,24 +1,91 @@
...
+                push!(ret,2)
+            end
+            push!(ret,3)
+        end
+        push!(ret, 4)
+        ret
+    end
+
+    @test nested() == 1:4
+    @test_broken @elide(nested, nested()) == [4]
```
Comment (simeonschaub) [inline test/runtests.jl]: Ah, yes! I really need to fix this.

File: test/runtests.jl [hunk 2]
```diff
@@ -1,24 +1,91 @@
...
+end
 
 using Documenter
 DocMeta.setdocmeta!(OptionalArgChecks, :DocTestSetup, :(using OptionalArgChecks); recursive=true)
 doctest(OptionalArgChecks; manual = false)
+
```
Comment (simeonschaub) [inline test/runtests.jl]: Was this intended?

**Review by simeonschaub — APPROVED** [review]
Thanks a lot for this! These test cases are definitely helpful, I really need to fix the nested one.

File: test/runtests.jl (hunk 1 above)
Comment (jw3126) [reply test/runtests.jl]: I gave it a stab, see below.

File: test/runtests.jl (hunk 2 above)
Comment (jw3126) [reply test/runtests.jl]: No not intended, good catch.

**Comment by simeonschaub** [comment]
Awesome! Thanks for this PR!

**Pull request merged**